plugins/Name_ShouldBeHousenumber.py

```diff
@@ -0,0 +1,77 @@
+#-*- coding: utf-8 -*-
+
+###########################################################################
+##                                                                       ##
+## Copyrights arrival-spring 2022                                        ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+from modules.OsmoseTranslation import T_
+from plugins.Plugin import Plugin
+import regex as re
+
+class Name_ShouldBeHousenumber(Plugin):
+
+    def init(self, logger):
+        Plugin.init(self, logger)
+        self.errors[804] = self.def_class(item = 5010, level = 1, tags = ['name', 'fix:chair'],
+            title = T_('Name with only numbers'),
+            detail = T_(
+'''This building is tagged with a name which contains only numbers
+and does not appear to have a main feature key.
+ For the majority of buildings this should be tagged as the housenumber.'''),
+            trap = T_(
+'''While uncommon, it is possible for a name to be only numbers.
+ This is particularly the case for some brands or amenities.''')
+        )
+        self.Numerical = re.compile(r"^[0-9]+$")
+        self.feature_keys = ["amenity", "craft", "emergency", "leisure", "military", "office", "railway", "shop", "tourism"]
+
+    def node(self, data, tags):
+        err = []
+        if "building" in tags:
+            if not any(feature in tags for feature in self.feature_keys):
+                if "name" in tags:
+                    if self.Numerical.match(tags["name"]):
+                        err.append({"class": 804, "text": T_("Concerns tag: `{0}`", '='.join(['name', tags['name']])) })
```

Comment:
A fix suggestion could be done in this case. Remove `name`, add `housenumber`.
https://github.com/osm-fr/osmose-backend/blob/master/doc/0-Index.md#fix-suggestion
 Eg: https://github.com/osm-fr/osmose-backend/blob/master/plugins/TagFix_Pharmacy.py#L46